### ISSUE TITLE
Missing RBAC for webhook?

### ISSUE THREAD

**Issue by johanneskastl** [vadimkim/cert-manager-webhook-hetzner#35]
I used this for about a year, and now I noticed that I cannot get new certificates.

```
0222 18:02:29.828161       1 controller.go:167] cert-manager/challenges "msg"="re-queuing item due to error processing" "error"="XXX.YYY is forbidden: User \"system:serviceaccount:cert-manager:cert-manager\" cannot create resource \"XXX\" in API group \"YYY\" at the cluster scope" "key"="AAA/BBB-zh49m-3528166467-3830210674"
```

Not sure if I missed some changes during upgrades, or if something in Kubernetes needs special treatment now.

From what I found this might (or not...) be related to missing RBAC permissions. [This issue](https://github.com/cert-manager/cert-manager/issues/3432) seems similar. 
So does [this one](https://github.com/Dopingus/cert-manager-webhook-dynu/issues/1), but that had another root cause (that I checked, my groupName is the same everywhere).

Any ideas?

**Comment by ibotty** [comment]
I also ran into this problem.  I don't know what changed (I assume it might be `certmanager` running as different service account), but I fixed it by setting `certManager,serviceAccountName=cert-manager` (it defaults to `cert-manager-controller` afair).

**Comment by johanneskastl** [comment]
> I also ran into this problem. I don't know what changed (I assume it might be `certmanager` running as different service account), but I fixed it by setting `certManager.serviceAccountName=cert-manager` (it defaults to `cert-manager-controller` afair).

Thanks for the reply. That seems to have done the trick.

**Comment by johanneskastl** [comment]
I created a PR to change the serviceAccountName, see #36. Not sure if this only hit us for whatever reasons, or if this should be fixed in general by the PR.

**Comment by pauvos** [comment]
Had the same issue. Works after chaning the serviceAccountName

**Comment by vadimkim** [comment]
I will add golang patch and make new release. Thanks!

**Comment by egandro** [comment]
More RBAC issues

https://github.com/vadimkim/cert-manager-webhook-hetzner/pull/37

The hook can't access flowcontrol-solver

I am using k8s:  v1.26.1+k3s1 (installed via k3s)

**Comment by egandro** [comment]
More more RBAC issues

I needed to add this roles to my cluster to cert manager: https://stackoverflow.com/a/73750470

**Comment by vadimkim** [comment]
Released at 1.2.2
